rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -0,0 +1,393 @@
+        second_link_asset = None
+        for asset in self.routing_assets:
+            if asset != to_asset:
+                found_second_link = self._find_pool_for(
+                    asset=to_asset,
+                    link_asset=asset,
+                    path=output,
+                )
+                if found_second_link:
+                    second_link_asset = asset
+                    break
+
+        if not found_second_link:
+            return []
+
+        # finally find the step of glue asset A <2nd link> glue asset B
+        assert second_link_asset is not None
+        assert link_asset is not None
+        pools = self.get_pool(link_asset, second_link_asset)
+        for pool in pools:
+            if pool != ZERO_ADDRESS:
+                output.insert(1, pool)
+                return output
+
+        return []
+
+    def get_price(
+        self,
+        from_asset: Asset,
+        to_asset: Asset,
+        block_identifier: BlockIdentifier,
+    ) -> Price:
+        """
+        Return the price of from_asset to to_asset at the block block_identifier.
+        External oracles are used if non eth tokens are used.
+
+        Can raise:
+        - PriceQueryUnsupportedAsset
+        - RemoteError
+        """
+        log.debug(
+            f'Searching price for {from_asset} to {to_asset} at '
+            f'{block_identifier!r} with {self.get_oracle_name()}',
+        )
+
+        # Use weth internally
+        if from_asset == A_ETH:
+            from_asset = A_WETH
+        if to_asset == A_ETH:
+            to_asset = A_WETH
+
+        if from_asset == to_asset:
+            return Price(ONE)
+
+        if not (from_asset.is_eth_token() and to_asset.is_eth_token()):
+            raise PriceQueryUnsupportedAsset(
+                f'Neither {from_asset} nor {to_asset} are ethereum tokens for the uniswap oracle',
+            )
+
+        # Could be that we are dealing with ethereum tokens as intances of Asset instead of
+        # EthereumToken, handle the conversion
+        from_asset_raw: Union[Asset, EthereumToken] = from_asset
+        to_asset_raw: Union[Asset, EthereumToken] = to_asset
+        if not isinstance(from_asset, EthereumToken):
+            from_as_token = EthereumToken.from_identifier(from_asset_raw.identifier)
+            if from_as_token is None:
+                raise PriceQueryUnsupportedAsset(f'Unsupported asset for uniswap {from_asset_raw}')
+            from_asset = from_as_token
+        if not isinstance(to_asset, EthereumToken):
+            to_as_token = EthereumToken.from_identifier(to_asset_raw.identifier)
+            if to_as_token is None:
+                raise PriceQueryUnsupportedAsset(f'Unsupported asset for uniswap {to_asset_raw}')
+            to_asset = to_as_token
+
+        assert isinstance(from_asset, EthereumToken), f'Got non valid token {from_asset}'
+        assert isinstance(to_asset, EthereumToken), f'Found non valid token {to_asset}'
```

Comment:
Please don't have assertions here. I guess you have them for type errors? What you should do instead is use the `from_as_token` and `to_as_token` which should have already been checked above for existing.